File: Python/contains-duplicate.py
class Solution:
    def containsDuplicate1(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        d = dict()

        for index, value in enumerate(nums):
            if value in d:
                return True
            d[value] = index

        return False

    def containsDuplicate2(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        return len(nums) > len(set(nums))

File: Swift/contains-duplicate.swift
class Solution {
    func containsDuplicate1(_ nums: [Int]) -> Bool {
        var dict = [Int: Int]()

        for (index, value) in nums.enumerated() {
            if let _ = dict[value] {
                return true
            }
            dict[value] = index
        }

        return false
    }

    func containsDuplicate2(_ nums: [Int]) -> Bool {
        return nums.count > Set(nums).count
    }
}
